Sleef issue when compiling master on Power9

I have been waiting the integration of SIMD instruction for IBM power processor....
Unfortunately, SLEEF does not compile on our computers (I checked LLVM-9 and LLVM-10)

I am not sure I can do much more to help, unless there is an obvious mistake I made.

```
[  0%] Building SLEEF to LLVM bitcode /home/kieffer/workspace/pocl/debug/lib/kernel/host/pwr9/c_sleef_glue_auto.c.bc
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:85:8: error: conflicting types for 'Sleef_sin_u35'
double Sleef_sin_u35 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:124:21: note: previous declaration is here
IMPORT CONST double Sleef_sin_u35(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:86:8: error: conflicting types for 'Sleef_cos_u35'
double Sleef_cos_u35 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:125:21: note: previous declaration is here
IMPORT CONST double Sleef_cos_u35(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:87:15: error: conflicting types for 'Sleef_sincos_u35'
Sleef_double2 Sleef_sincos_u35 (double);
              ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:126:28: note: previous declaration is here
IMPORT CONST Sleef_double2 Sleef_sincos_u35(double);
                           ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:88:8: error: conflicting types for 'Sleef_tan_u35'
double Sleef_tan_u35 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:127:21: note: previous declaration is here
IMPORT CONST double Sleef_tan_u35(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:89:8: error: conflicting types for 'Sleef_asin_u35'
double Sleef_asin_u35 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:128:21: note: previous declaration is here
IMPORT CONST double Sleef_asin_u35(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:90:8: error: conflicting types for 'Sleef_acos_u35'
double Sleef_acos_u35 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:129:21: note: previous declaration is here
IMPORT CONST double Sleef_acos_u35(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:91:8: error: conflicting types for 'Sleef_atan_u35'
double Sleef_atan_u35 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:130:21: note: previous declaration is here
IMPORT CONST double Sleef_atan_u35(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:92:8: error: conflicting types for 'Sleef_atan2_u35'
double Sleef_atan2_u35 (double, double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:131:21: note: previous declaration is here
IMPORT CONST double Sleef_atan2_u35(double, double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:93:8: error: conflicting types for 'Sleef_log_u35'
double Sleef_log_u35 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:132:21: note: previous declaration is here
IMPORT CONST double Sleef_log_u35(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:94:8: error: conflicting types for 'Sleef_cbrt_u35'
double Sleef_cbrt_u35 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:133:21: note: previous declaration is here
IMPORT CONST double Sleef_cbrt_u35(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:95:8: error: conflicting types for 'Sleef_sin_u10'
double Sleef_sin_u10 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:134:21: note: previous declaration is here
IMPORT CONST double Sleef_sin_u10(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:96:8: error: conflicting types for 'Sleef_cos_u10'
double Sleef_cos_u10 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:135:21: note: previous declaration is here
IMPORT CONST double Sleef_cos_u10(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:97:15: error: conflicting types for 'Sleef_sincos_u10'
Sleef_double2 Sleef_sincos_u10 (double);
              ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:136:28: note: previous declaration is here
IMPORT CONST Sleef_double2 Sleef_sincos_u10(double);
                           ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:98:8: error: conflicting types for 'Sleef_tan_u10'
double Sleef_tan_u10 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:137:21: note: previous declaration is here
IMPORT CONST double Sleef_tan_u10(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:99:8: error: conflicting types for 'Sleef_asin_u10'
double Sleef_asin_u10 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:138:21: note: previous declaration is here
IMPORT CONST double Sleef_asin_u10(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:100:8: error: conflicting types for 'Sleef_acos_u10'
double Sleef_acos_u10 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:139:21: note: previous declaration is here
IMPORT CONST double Sleef_acos_u10(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:101:8: error: conflicting types for 'Sleef_atan_u10'
double Sleef_atan_u10 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:140:21: note: previous declaration is here
IMPORT CONST double Sleef_atan_u10(double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:102:8: error: conflicting types for 'Sleef_atan2_u10'
double Sleef_atan2_u10 (double, double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:141:21: note: previous declaration is here
IMPORT CONST double Sleef_atan2_u10(double, double);
                    ^
In file included from /home/kieffer/workspace/pocl/lib/kernel/sleef/libm/sleef_glue_auto.c:3:
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef_cl.h:103:8: error: conflicting types for 'Sleef_log_u10'
double Sleef_log_u10 (double);
       ^
/home/kieffer/workspace/pocl/lib/kernel/sleef/include/sleef.h:142:21: note: previous declaration is here
IMPORT CONST double Sleef_log_u10(double);
                    ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [lib/kernel/host/CMakeFiles/kernel_host_pwr9.dir/build.make:2778: lib/kernel/host/pwr9/c_sleef_glue_auto.c.bc] Error 1
make[1]: *** [CMakeFiles/Makefile2:929: lib/kernel/host/CMakeFiles/kernel_host_pwr9.dir/all] Error 2
make: *** [Makefile:182: all] Error 2
```

--- a/README.PPC64le
+++ b/README.PPC64le
@@ -1,0 +1,28 @@
+PoCL is used to provide OpenCL on IBM AC922 computers 
+featuring IBM Power9 processors and Nvidia Tesla V100 GPU 
+interconnected with NVlink v2 (up to 72 GByte/s). 
+This has been tested under debian_11 and Ubuntu_20.04.
+
+Officially, Nvidia does not support OpenCL on this platform 
+and the driver they are shipping is lacking the compiler part.
+
+## Building tricks (as of 04/2023):
+
+The PPC64le features 128-bit vector unit (Altivec/VSX) which 
+are easily confused by the C++ compiler with the C++ vector 
+instruction when using the compile option `-std=c++XX`.
+The corresponding code usually fails compiling.
+The trick is to pass the option `-std=gnu++XX`.
+
+For example, when configuring pocl:
+```
+cmake .. -DLLVM_CXXFLAGS="-std=gnu++14 ..." -DENABLE_CUDA=ON
+```
+
+The full list of  options for the CXXFLAGS is obtained with:
+```
+llvm-config --cxxflags|sed -e "s/std=c/std=gnu/"
+```
+Later on, the build continues with `make` ...
+
+Note the CUDA option to enable the GPU support on those computers.

--- a/README.md
+++ b/README.md
@@ -76,6 +76,12 @@ from this PPA: https://launchpad.net/~ocl-icd/+archive/ubuntu/ppa
 Additionally, if you want the CPU device to report as 3.0 OpenCL
 you will need LLVM 14 or newer.
 
+### GPU support on different architectures
+
+PoCL can be used to provide OpenCL driver on several architectures where the hardware manufacturer does not ship them 
+like Nvidia Tegra (ARM) or IBM Power servers. On PPC64le servers, there are specific instructions to handle the build 
+of PoCL in [README.PPC64le](./README.PPC64le). 
+
 ### Windows
 
 Windows support has been unmaintained for a long time and building on Windows

